frontend/server/src/Controllers/Problem.php

```diff
@@ -164,10 +168,12 @@ private static function validateCreateOrUpdate(
                 'visibility',
                 [\OmegaUp\Controllers\Problem::VISIBILITY_PRIVATE, \OmegaUp\Controllers\Problem::VISIBILITY_PUBLIC]
             );
-            $r['selected_tags'] = json_decode($r['selected_tags']);
-            if (!empty($r['selected_tags'])) {
-                foreach ($r['selected_tags'] as $tag) {
-                    if (empty($tag->tagname)) {
+            /** @var \OmegaUp\Tag[]|null */
+            $selectedTags = json_decode($r['selected_tags']);
```

Comment:
```suggestion
            $selectedTags = json_decode($r['selected_tags'], /*assoc=*/true);
```

creo que por eso se estaba enojando psalm. y no puedes declarar esto como `\OmegaUp\Tag[]|null` porque lo que regresa `json_decode` es una objeto de una clase artificial que no se puede convertir a ninguna otra clase.

Reply:
Listo